Adding PR template (#5139)
microsoft/aspire: .github/pull_request_template.md

## Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Checklist

- Is this feature complete or do we expect follow-up changes before this code can ship?
  - [ ] Yes
  - [ ] No
- Are you including unit tests for the changes and scenario tests if relevant?
  - [ ] Yes
  - [ ] No
- Did you add public API? Did you have an API Review for it?
  - [ ] Yes
  - [ ] No
- Does the change make any security assumptions or guarantees? Have you done a threat model and had a security review?
  - [ ] Yes

>   - [ ] No

Reviewer: For these 2 lines, they both have 2 questions and the answer can be different for each question. How would you answer that this changes public API but doesn't have an API Review (yet)?

Author: Good feedback, I'll fix this.

## Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Checklist

> - Is this feature complete or do we expect follow-up changes before this code can ship?
>   - [ ] Yes
>   - [ ] No

Reviewer 2: This is a confusing Yes/No question. What are you saying yes to?
```suggestion
- Is this feature complete?
  - [ ] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
```

- Are you including unit tests for the changes and scenario tests if relevant?
  - [ ] Yes
  - [ ] No

> - Did you add public API?

Reviewer 3: Can we also ask whether the public API has XML doc comments including:

1. `<remarks />` if greater explanation is required than what can be provided in the summary.
2. `<example />` elements for any examples with code placed in a `<code />` element.

  - [ ] Yes

>     - If yes, did you have an API Review for it?
>       - [ ] Yes
>       - [ ] No
>   - [ ] No

Author: ```suggestion
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [ ] No
  - [ ] No
```
